### PR TITLE
Simplifies snackbar mechanism

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/SnackbarUtils.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import android.support.design.widget.Snackbar;
+import android.view.View;
+
+/**
+ * Provides a method to show a Snackbar.
+ */
+public class SnackbarUtils {
+
+    public static void showSnackBar(View v, String snackbarText) {
```
Comment (florina-muntenescu) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/SnackbarUtils.java]: java doc for the method

File: todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModelTest.java [hunk 2]
```diff
@@ -96,4 +98,25 @@ public void populateTask_callsRepoAndUpdatesView() {
         assertThat(mAddEditTaskViewModel.title.get(), is(testTask.getTitle()));
         assertThat(mAddEditTaskViewModel.description.get(), is(testTask.getDescription()));
     }
+
+    @Test
```
Comment (florina-muntenescu) [inline todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModelTest.java]: In all tests we add comments for every step. can you do that for both of these tests?
Also, add a new line between the act and the assert for both tests

File: todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModelTest.java [hunk 3]
```diff
@@ -179,4 +184,21 @@ private void setupViewModelRepositoryCallback() {
 
         mGetTaskCallbackCaptor.getValue().onTaskLoaded(mTask); // Trigger callback
     }
+
+    @Test
+    public void updateSnackbar_nullValue() {
```
Comment (florina-muntenescu) [inline todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModelTest.java]: remove this new line
add more comments
add a new line between the act and assert
(same for the other test)

File: todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModelTest.java [hunk 4]
```diff
@@ -214,4 +218,21 @@ public void getTasksAddViewVisible() {
         // Then the "Add task" action is visible
         assertThat(mTasksViewModel.getTasksAddViewVisible(), is(true));
     }
+
+    @Test
+    public void updateSnackbar_nullValue() {
+        // Get a reference to the class under test
+
```
Comment (florina-muntenescu) [inline todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModelTest.java]: remove this new line
add more comments
add a new line between the act and assert
(same for the other test)

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.java [hunk 5]
```diff
@@ -116,7 +115,7 @@ public void setViewModel(TasksViewModel viewModel) {
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
-        setupSnackbar();
+        setupSnackBar();
```
Comment (florina-muntenescu) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.java]: you're using everywhere "Snackbar" (small b). keep it consistent

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/SnackbarUtils.java (hunk 1 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/SnackbarUtils.java]: Self-explanatory

File: todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModelTest.java (hunk 2 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModelTest.java]: Done

File: todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModelTest.java (hunk 3 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/test/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailViewModelTest.java]: Done

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.java (hunk 5 above)
Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.java]: Done

File: todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java [hunk 6]
```diff
@@ -63,12 +63,17 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
         setupFab();
 
-        setupSnackbar();
+        setupSnackBar();
```
Comment (florina-muntenescu) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java]: `setupSnackbar`

Comment (JoseAlcerreca) [inline todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailFragment.java]: Done

**Comment by JoseAlcerreca** [comment]
Tests pass, something wrong with Firebase Test lab's emulator.

**Pull request merged**